File: src/LiftAndShift.Core/Workouts/WorkoutLifts.cs
using LiftAndShift.Core.Lifts;
using LiftAndShift.Core.ProgrammeAggregate;

namespace LiftAndShift.Core.Workouts;

public static class WorkoutLifts
{
  public static IReadOnlyList<Lift> For(Workout workout, TrainingPhase trainingPhase)
  {
    if (workout == Workout.A)
    {
      return [Lift.Squat, Lift.Press, Lift.Deadlift];
    }

    if (workout == Workout.B)
    {
      return [Lift.Squat, Lift.BenchPress, PhaseLift(trainingPhase)];
    }

    throw new ArgumentOutOfRangeException(nameof(workout), workout, "Unknown workout.");
  }

  private static Lift PhaseLift(TrainingPhase trainingPhase) => trainingPhase.Value switch
  {
    1 => Lift.Deadlift,
    2 => Lift.Row,
    3 => Lift.LatPulldown,
    _ => throw new ArgumentOutOfRangeException(nameof(trainingPhase), trainingPhase, "Unknown Training Phase.")
  };
}

File: src/LiftAndShift.Web/ClientApp/app/utils/workoutLifts.ts
// Mirrors LiftAndShift.Core.Workouts.WorkoutLifts.For server-side: which lifts a workout means,
// given the family member's current Training Phase. Workout A never changes; Workout B's third lift
// depends on phase (see CONTEXT.md > Training Phase).
const PHASE_THIRD_LIFT: Record<number, string> = {
  1: 'Deadlift',
  2: 'Row',
  3: 'LatPulldown'
}

export function liftsForWorkout(workout: 'A' | 'B', trainingPhase: number): string[] {
  if (workout === 'A') {
    return ['Squat', 'Press', 'Deadlift']
  }

  return ['Squat', 'BenchPress', PHASE_THIRD_LIFT[trainingPhase]]
}
